app/frontend/actions.ts

```diff
@@ -1013,7 +1016,8 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
      * pool validation must happen before debouncing
      * but pool info shown after
      */
-    const poolHash = hash || e.target.value
+    const poolByTicker = e && e.target.value && state.validStakepoolsByTicker[e.target.value]
+    const poolHash = hash || (poolByTicker && poolByTicker.poolHash) || e.target.value
     const validationError = poolIdValidator(poolHash, state.validStakepools)
```

Comment:
I refactored this function in a slightly different way. It doesn't make much sense to forward the responsibility of handling the input event to actions. Now it's only taking in a `poolHash`.

Reply:
I totally agree, lets move the logic from actions to the actual component, in that case, tho, we should also do the validation in the component. As mentioned before, the search by ticker might not work and we need a specific error for that. So we can either move the validation to the component and pass the resulting error to action along with the poolhash or do it in actions as we used to.

Its crucial to handle this logic in one place, so either we should get the pool info in the component and pass everything to actions which would just set it or handle everything in actions

Reply:
I totally agree, lets move the logic from actions to the actual component, in that case, tho, we should also do the validation in the component. As mentioned before, the search by ticker might not work and we need a specific error for that. So we can either move the validation to the component and pass the resulting error to action along with the poolhash or do it in actions as we used to. 

Its crucial to handle this logic in one place, so either we should get the pool info in the component and pass everything to actions which would just set it or handle everything in actions

Reply:
I moved validation to component. I also changed behaviour of search. If we have tickers from BE, it works like previously. If we don't have tickers from BE, the label shows only `Stake Pool ID` instead of `Ticker or Stake Pool ID`, same change in principle for showing error.